src/integrations/prefect-kubernetes/prefect_kubernetes/worker.py

```diff
@@ -380,8 +380,30 @@ def prepare_for_flow_run(
                 preparation.
             flow: The flow associated with the flow run used for preparation.
         """
+        # Save original env if it's a list
+        original_env_list = None
+        if isinstance(self.env, list):
+            original_env_list = self.env.copy()
 
+        # Call parent method which will convert env to dict format
         super().prepare_for_flow_run(flow_run, deployment, flow, work_pool, worker_name)
+
+        # If we had a list format originally, we need to merge the dict format back into the list
+        if original_env_list is not None:
+            # Convert dict env back to list format, but preserve original list items
+            # Create a set of env var names from the original list to check for duplicates
+            original_env_names = {item["name"] for item in original_env_list}
+
+            # Convert the dict env to list format for new items
+            dict_env_as_list = [
+                {"name": k, "value": v}
+                for k, v in self.env.items()
+                if k not in original_env_names
+            ]
+
+            # Combine original list with new items
+            self.env = original_env_list + dict_env_as_list
```

Comment:
would it be easier to convert the list to a dictionary outright so that we don't have to remember to dispatch on list vs. dict down below?

Reply:
ah good point - updated in [96fb1e8](https://github.com/PrefectHQ/prefect/pull/17512/commits/96fb1e86cbda926cb085219d738c1b711a7611c0)

Reply:
I don't think this is any easier because you still have the special `valueFrom` case to check for either way, I'm leaning toward reverting to what I had before unless you disagree @cicdw